src/client.ts

```diff
@@ -5338,9 +5334,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         // There is no guarantee that the event ended up in "timeline" (we might have switched to a neighbouring
         // timeline) - so check the room's index again. On the other hand, there's no guarantee the event ended up
         // anywhere, if it was later redacted, so we just return the timeline we first thought of.
-        return timelineSet.getTimelineForEvent(eventId)
-            ?? timelineSet.room.findThreadForEvent(event)?.liveTimeline // for Threads degraded support
```

Comment:
`?? timelineSet.room.findThreadForEvent(event)?.liveTimeline // for Threads degraded support` was added in [#2261](https://github.com/matrix-org/matrix-js-sdk/pull/2261/files#diff-25d66d74617fe2e23d7946bd6e3ba95640ab1b9bc8947445d604fc271c7c1f12R5328). Is it necessary? I'm just trying to get the context behind why we have it.

What does `Threads degraded support` exactly map to?

If `supportsExperimentalThreads` is disabled all events will be added to the main timeline ([`eventShouldLiveIn` always returns `shouldLiveInRoom: true`](https://github.com/matrix-org/matrix-js-sdk/blob/5112340040dbb09e5d9b71cdec183d87ec17638e/src/models/room.ts#L1718-L1720))

Reply:
> Threads degraded support

Server has no APIs to help the client so the client has to build the threads objects without filters & relations support on the server

Reply:
Thanks for the context!

I think it's safe to drop this case given `getTimelineForEvent` is always used in a fire and forget manner, see https://github.com/matrix-org/matrix-js-sdk/pull/2521#discussion_r920214132